101-octopus-deploy-tentacle-windows/azuredeploy.json

```diff
@@ -0,0 +1,335 @@
+{
+  "$schema": "http://schema.management.azure.com/schemas/2014-04-01-preview/deploymentTemplate.json",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "vmAdminUsername": {
+      "type": "string",
+      "metadata": {
+        "description": "Admin username for the Virtual Machine."
+      }
+    },
+    "vmAdminPassword": {
+      "type": "securestring",
+      "metadata": {
+        "description": "Admin password for the Virtual Machine."
+      }
+    },
+    "vmDnsName": {
+      "type": "string",
+      "metadata": {
+        "description": "Unique DNS Name for the Public IP used to access the Virtual Machine."
+      }
+    },
+    "vmSize": {
+      "defaultValue": "Standard_D2_v2",
+      "type": "string",
+      "metadata": {
+        "description": "Size of the Virtual Machine"
+      }
+    },
+    "tentacleOctopusServerUrl": {
+      "type": "string",
+      "metadata": {
+        "description": "The URL of the octopus server with which to register"
+      }
+    },
+    "tentacleApiKey": {
+      "type": "securestring",
+      "metadata": {
+        "description": "The Api Key to use to register the Tentacle with the server"
+      }
+    },
+    "tentacleCommunicationMode": {
+      "defaultValue": "Listen",
+      "allowedValues": [
+        "Listen",
+        "Poll"
+      ],
+      "type": "string",
+      "metadata": {
+        "description": "The type of Tentacle - whether the Tentacle listens for requests from server, or actively polls the server for requests"
+      }
+    },
+    "tentaclePort": {
+      "defaultValue": 10933,
+      "minValue": 0,
+      "maxValue": 65535,
+      "type": "int",
+      "metadata": {
+        "description": "The port on which the Tentacle should listen, when CommunicationMode is set to Listen, or the port on which to poll the server, when CommunicationMode is set to Poll. By default, Tentacle's listen on 10933 and poll the Octopus Server on 10943."
+      }
+    },
+    "tentacleRoles": {
+      "type": "string",
+      "metadata": {
+        "description": "A comma delimited list of roles to apply to the Tentacle"
```

Comment:
Capitalise Roles in this case I think. Can you provide links to our docs? (Probably not)